core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -409,7 +409,12 @@ public static OAuthTokenResponse tokenResponseFromJson(JsonNode json) {
             .withIssuedTokenType(JsonUtil.getStringOrNull(ISSUED_TOKEN_TYPE, json));
 
     if (json.has(EXPIRES_IN)) {
-      builder.setExpirationInSeconds(JsonUtil.getInt(EXPIRES_IN, json));
+      // Some IdPs like Entra ID return the expiration time as a string
+      try {
+        builder.setExpirationInSeconds(JsonUtil.getInt(EXPIRES_IN, json));
```

Comment:
the issue with this change is that the Spec states that this should be an integer: https://github.com/apache/iceberg/blob/4b52dbd89bf583f1bbdbf0b7a73b62404812ee11/open-api/rest-catalog-open-api.yaml#L3787-L3790

I think we should discuss on the DEV ML what we'd like to do in such a case, because this opens the codebase for having a precedent where we're less strict about what we're expecting in terms of value types